meemoo/services.py

```diff
@@ -92,21 +92,46 @@ def get_pid(self) -> str:
         return pid
 
 
+class OrgApiError(Exception):
+    pass
+
+
 class OrganisationsService(Service):
     """ Abstraction for the organisations-api. """
 
     def __init__(self, ctx):
         self.name = "organisations-api"
         super().__init__(ctx)
 
-    def get_organisation(self, or_id):
+    def _construct_query(self, or_id: str):
+        """Construct the Graphql query to retrieve the CP name defined in the MAM
+        Args:
+            or_id: The cp-id for which to retrieve the CP name.
+        Returns:
+            The graphql query.
+        """
+        query = f"""{{
+            organizations(id:"{or_id}") {{
+                mam_label
+            }}
+        }}"""
+        return query
+
+    def get_mam_label(self, or_id):
         # Make sure the OR is uppercase. Organisation api needs it.
         or_id = or_id[:2].upper() + or_id[2:]
 
-        response = requests.get(f"{self.host}org/{or_id}")
-        organisation = response.json()["data"]
-        return organisation
-
+        query = self._construct_query(or_id)
+        data_payload = {"query": query}
+        response = requests.post(
+            self.host,
+            json=data_payload,
+        )
+        try:
+            mam_label = response.json()["data"]["organizations"][0]["mam_label"]
```

Comment:
We should probably not get back more than one result...

Reply:
We should get one or zero result(s) back. If we do get more than one result back then there is an issue with the org API. In that case we just use the "first" one returned. An other option would be to stop handling the event and take appropriate action (log, nack, ... ?).